File: src/lib/nodejs/logger.js
/**
 * Logging utilities for standardized logging across scripts.
 */

const fs = require('fs');
const path = require('path');

/**
 * Logger class for standardized logging.
 */
class Logger {
    /**
     * Create a logger instance.
     * @param {string} name - Logger name
     * @param {string|null} logFile - Optional log file name (will be created in output/logs/)
     */
    constructor(name, logFile = null) {
        this.name = name;
        this.logFile = logFile;

        if (logFile) {
            const logDir = path.join(__dirname, '..', '..', '..', 'output', 'logs');
            if (!fs.existsSync(logDir)) {
                fs.mkdirSync(logDir, { recursive: true });
            }
            this.logPath = path.join(logDir, logFile);
        }
    }

    /**
     * Log a message with specified level.
     * @param {string} level - Log level (INFO, ERROR, DEBUG, etc.)
     * @param {string} message - Message to log
     */
    log(level, message) {
        const timestamp = new Date().toISOString();
        const logMessage = `${timestamp} - ${this.name} - ${level} - ${message}`;

        console.log(logMessage);

        if (this.logFile) {
            fs.appendFileSync(this.logPath, logMessage + '\n');
        }
    }

    /**
     * Log info message.
     * @param {string} message - Message to log
     */
    info(message) {
        this.log('INFO', message);
    }

    /**
     * Log error message.
     * @param {string} message - Message to log
     */
    error(message) {
        this.log('ERROR', message);
    }

    /**
     * Log debug message.
     * @param {string} message - Message to log
     */
    debug(message) {
        this.log('DEBUG', message);
    }
}

module.exports = { Logger };

File: src/lib/python/logger.py
"""Logging utilities for standardized logging across scripts."""

import logging
from pathlib import Path


def setup_logger(name, log_file=None, level=logging.INFO):
    """Set up standardized logger.

    Args:
        name: Logger name (usually __name__)
        log_file: Optional log file name (will be created in output/logs/)
        level: Logging level (default: INFO)

    Returns:
        Configured logger instance
    """
    logger = logging.getLogger(name)
    logger.setLevel(level)

    # Avoid adding handlers multiple times
    if logger.handlers:
        return logger

    # Console handler
    console_handler = logging.StreamHandler()
    console_handler.setLevel(logging.INFO)
    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
    console_handler.setFormatter(formatter)
    logger.addHandler(console_handler)

    # File handler
    if log_file:
        log_path = Path(__file__).parent.parent.parent.parent / "output" / "logs" / log_file
        log_path.parent.mkdir(parents=True, exist_ok=True)
        file_handler = logging.FileHandler(log_path)
        file_handler.setLevel(logging.DEBUG)
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)

    return logger
